core/transport/src/grpc/block_master.proto

```diff
@@ -134,6 +140,11 @@ service BlockMasterClientService {
    */
   rpc GetUsedBytes(GetUsedBytesPOptions) returns (GetUsedBytesPResponse);
 
+  /**
+   * Returns a list of workers addresses.
+   */
+  rpc GetWorkerAddresses(GetWorkerAddressesPOptions) returns (GetWorkerAddressesPResponse);
```

Comment:
The previous getWorkerInfoList() requires the locking of each worker to get the full worker info list. In most cases (like the places i change from getWorkerInfoList() -> getWorkerAddresses()), only worker addresses are needed. Worker addresses are initialized when creating the MasterWorkerInfo and never changed. No locks are needed to get the worker addresses.

If we find `getWorkerInfoList()` API takes longer than expected because it's locking mechanism, add a `getWorkerAddresses()` can help reduce the locks needed and reduce contention.